File: projects/order-creation-queries/core/src/main/java/com/robinfood/core/mappers/CouponMappers.java
package com.robinfood.core.mappers;

import com.robinfood.core.dtos.CouponDTO;
import com.robinfood.core.entities.CouponEntity;

public final class CouponMappers {

    private CouponMappers() {
        // this constructor is empty because it is a utility class
    }

    public static CouponDTO toCouponsDTO(CouponEntity couponEntity) {

        return CouponDTO.builder()
                .value(couponEntity.getValue())
                .code(couponEntity.getCode())
                .build();
    }
}

File: projects/posv2/core/src/main/kotlin/com/robinfood/core/mappers/CouponMappers.kt
package com.robinfood.core.mappers

import com.robinfood.core.dtos.transactionrequest.CouponDTO
import com.robinfood.core.entities.transactionrequest.CouponEntity

fun CouponDTO.toCouponEntity(): CouponEntity {
    return CouponEntity(
            code = code,
            value = value
    )
}